app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+class PostsController < ApplicationController
+  before_action :sign_up_if_not_logged_in
+  def index
+    @posts = Post.all
+  end
+
+  def show
+    @post = Post.find(params[:id])
+  end
+
+  def new
+    @post = current_user.posts.build
+  end
+
+  def create
+    @post = current_user.posts.build(post_params)
+    if @post.save
+      redirect_to root_path
+    else
+      redirect_to new_post_path
+    end
+  end
+
```

Comment:
**Optional**
It would be nice if you add a way to delete posts. This will add value to your project since it will make more similar to real-world projects

Reply:
Thanks for comment we will implement delete post option in the future milestone.